pkg/deploy/registry/instantiate/rest.go

```diff
@@ -0,0 +1,286 @@
+package instantiate
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+func NewREST(store registry.Store, oc client.Interface, kc kclient.Interface, codec runtime.Codec) *REST {
+	store.UpdateStrategy = Strategy
+	return &REST{store: &store, isn: oc, rn: kc, codec: codec}
+}
+
+// REST implements the Updater interface.
+var _ = rest.Updater(&REST{})
+
+type REST struct {
+	store *registry.Store
+	isn   client.ImageStreamsNamespacer
+	rn    kclient.ReplicationControllersNamespacer
+	codec runtime.Codec
+}
+
+func (s *REST) New() runtime.Object {
+	return &deployapi.DeploymentRequest{}
+}
+
+// Update instantiates a deployment config
+func (r *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
+	configObj, err := r.store.Get(ctx, name)
+	if err != nil {
+		return nil, false, err
+	}
+	config := configObj.(*deployapi.DeploymentConfig)
+
+	old := deployapi.RequestForConfig(config)
+	obj, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return nil, false, err
+	}
+
+	req, ok := obj.(*deployapi.DeploymentRequest)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to DeploymentRequest update: %#v", obj))
+	}
+
+	if errs := validation.ValidateDeploymentRequest(req); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(deployapi.Kind("DeploymentRequest"), req.Name, errs)
+	}
+
+	// TODO: Consider instantiating paused deployments.
+	if config.Spec.Paused {
+		err = errors.NewBadRequest(fmt.Sprintf("deployment config %q is paused - run 'oc rollout resume dc/%s' to unpause", config.Name, config.Name))
+		return nil, false, err
+	}
+
+	updated := false
+	// We need to process the deployment config before we can determine if it is possible to trigger
+	// a deployment.
+	if req.Latest {
+		updated, err = r.Process(config, req.Force)
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	canTrigger, causes, err := r.CanTrigger(config, req.Force)
+	if err != nil {
+		return nil, false, err
+	}
+	if canTrigger {
+		config.Status.Details = new(deployapi.DeploymentDetails)
```

Comment:
> Why the strategy I am using is not ok?

We don't want to go direct to etcd and at some point we want to prevent updaters of a DC from touching status.